src/shims/os_str.rs

```diff
@@ -62,27 +62,28 @@ fn convert_path_separator<'a>(
 
 impl<'mir, 'tcx: 'mir> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
 pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+
+    #[cfg(unix)]
+    fn bytes_to_os_str<'a>(&self, bytes: &'a [u8]) -> InterpResult<'tcx, &'a OsStr> {
```

Comment:
Why did you add a `self` parameter?

Reply:
This makes it easier to invoke from outside this module - if it's an inherent or freestanding method, you need to name this module or this `EvalContextExt` trait or explicitly.

Reply:
That's what we do for other context-independent helper functions though, so I think we should also do it here.

Reply:
Done